pymeasure/experiment/parameters.py

```diff
@@ -46,6 +46,11 @@ class Parameter:
 
     def __init__(self, name, default=None, ui_class=None, group_by=None, group_condition=True):
         self.name = name
+        separator = ": "
+        if separator in name:
+            raise ValueError("The provided name argument '%s' contains the sparator '%s')."
+                             % (name, separator))
```

Comment:
You could use the more modern f-string:
```suggestion
            raise ValueError(f"The provided name argument '{name}' contains the separator '{separator}'.")
```

Btw. you spelled it wrong: "sparator" (without "e"). And you have an additional parenthesis in the string (second to last character).

Reply:
Careless of me... Also, I like to use f-strings. Thanks for your comments.